master/pkg/protoutils/protoless/less.go

```diff
@@ -55,6 +55,23 @@ func CheckpointSearcherMetricLess(ai, aj *checkpointv1.Checkpoint) bool {
 	return ai.Training.SearcherMetric.Value < aj.Training.SearcherMetric.Value
 }
 
+// CheckpointMetricNameLess compares checkpoints by a metric name, falling back to
+// report time when equal. Order makes sure nulls are always last.
+func CheckpointMetricNameLess(ai, aj *checkpointv1.Checkpoint, metricName string) bool {
+	aiMetricValue, aiOk := ai.Training.ValidationMetrics.AvgMetrics.AsMap()[metricName]
+	ajMetricValue, ajOk := aj.Training.ValidationMetrics.AvgMetrics.AsMap()[metricName]
+	if ai.Training == nil || !aiOk {
```

Comment:
`ai.Training` check is useless because you already checked `ai.Training.ValidationMetrics`, so we would've panicked already. i think we just need to put some nil checks high. rip golang, i want c#'s / ts's `(ai?.Training?.ValidationMetrics?.AvgMetrics ?? {})[metricName]`

Reply:
changed to use protobuf getters